python/preamble.h

```diff
@@ -0,0 +1,4 @@
+namespace pybind11 { namespace detail {
+    template <typename T>
+    struct type_caster<boost::optional<T>> : optional_caster<boost::optional<T>> {};
+}}
```

Comment:
Can you please explain what's going on here?

Reply:
This is a type caster for boost::optional so we can use optionals in python. @yetongumich I think it's better that I handle this, so I'll close this. Sorry for that :)

Reply:
Got it. That sounds good!